### PR TITLE
Add show_categories() method, update show_filters(), and add '__all__' 

### PR DESCRIPTION
**Pull request by irahorecka**
Hi Julio,

This PR groups a few changes that I believe would be beneficial to the package.
The biggest feature is `show_categories()`, which displays valid categories for a given Craigslist class.
I update `show_filters()` to print Craigslist class and category (if provided) when displaying class-specific filters.
Lastly, I added `__all__` to only allow imports of public objects in the craigslist module when using `import *`.

Let me know what you think. If these features look OK, I can follow up with an update of README.

-Ira

### CODE REVIEW

File: craigslist/base.py [hunk 1]
```diff
@@ -410,13 +410,35 @@ def get_list_filters(cls, url):
...
+        response = utils.requests_get(url)
+        soup = utils.bs(response.content)
+        cats = [
+            input.get("data-abb")
+            for input in soup.find_all("input", {"class": "catcheck multi_checkbox"})
+        ]
+        cats_title = [a.contents[0] for a in soup.find_all("a", {"class": "category"})]
+
+        print("%s categories:" % cls.__name__)
+        for cat, cat_title in iteritems(dict(zip(cats, cats_title))):
```
Comment (juliomalegria) [inline craigslist/base.py:428]: Wouldn't it be the same to do `zip(cats, cats_title)` without the `dict+iteritems`? Or maybe I'm missing something. Maybe we can add a `sorted` to sort alphabetically.

File: craigslist/base.py [hunk 2]
```diff
@@ -410,13 +410,35 @@ def get_list_filters(cls, url):
...
         print('Base filters:')
         for key, options in iteritems(cls.base_filters):
             value_as_str = '...' if options['value'] is None else 'True/False'
             print('* %s = %s' % (key, value_as_str))
-        print('Section specific filters:')
+
+        if category is None:
+            print("\n%s filters:" % cls.__name__)
+        else:
+            print("\n%s filters with category '%s':" % (cls.__name__, category))
```
Comment (juliomalegria) [inline craigslist/base.py:441]:
```suggestion
            print("\n%s filters for category '%s':" % (cls.__name__, category))
```
Sounds a bit better IMO.

**Review by juliomalegria — COMMENTED** [review]
Love this changes! Thanks a lot @irahorecka. I've left a couple of nit-picky comments. If you want you can update the README here, otherwise I can do it after merging this.

File: craigslist/base.py (hunk 1 above)
Comment (irahorecka) [reply craigslist/base.py:428]: Whoops you are totally right! Brain fart on my end

**Comment by juliomalegria** [comment]
Merged! Thanks @irahorecka :)

**Pull request merged**